tests/test_dataclasses.py

```diff
@@ -489,3 +489,43 @@ class TestClassVar:
 
     tcv = TestClassVar(2)
     assert tcv.klassvar == "I'm a Class variable"
+
+
+def test_unsupported_field_type():
+    with pytest.raises(TypeError):
```

Comment:
```suggestion
    with pytest.raises(TypeError) as exc_info:
```
then at least one test should check `str(exc_info.value)` is correct.

Reply:
`pytest.raises` allows you to check that it matches a regex, which seems a little cleaner. Is it okay to do it that way?

Reply:
okay.